### ISSUE TITLE
GitLab Provider Should Skip Nonce Validation on ID Token by Default

### ISSUE THREAD

**Issue by michaeljfazio** [oauth2-proxy/oauth2-proxy#2275]
## Expected Behavior

GitLab provider should skip nonce validation for id tokens when executing token refresh flow.

## Current Behavior

GitLab provider attempts to validate id token nonce on token refresh. Nonce validation will always fail, preventing session from ever refreshing. Setting insecure skip nonce option to true has no effect. 

## Possible Solution

Default [```InsecureSkipNonce```](https://github.com/oauth2-proxy/oauth2-proxy/blob/b78c391adc6da2598870f19469d18ca1ab035897/providers/gitlab.go#L43C67-L43C67) to ```True``` for GitLab Provider and implement ability to override with ```--insecure-oidc-skip-nonce``` value.

## Steps to Reproduce (for bugs)

1. Login with GitLab provider.
2. Wait for refresh flow execution to occur on access token timeout.
3. Refresh of token fails due to invalid nonce.
```
oauth2-proxy [2023/10/19 01:39:42] [oidc.go:117] nonce verification failed: id_token nonce claim does not match the session nonce
oauth2-proxy [2023/10/19 01:39:42] [stored_session.go:94] Error loading cookied session: error refreshing access token for session (Session{email ...
oauth2-proxy [2023/10/19 01:39:42] [oauthproxy.go:970] No valid authentication in request. Initiating login.
```

## Context

GitLab provider is unusable because access token timeout is 2m and refresh flow will always fail. Thus, the maximum amount of time a valid session can be sustained is < 2m.

## Your Environment

```
Image:         quay.io/oauth2-proxy/oauth2-proxy:v7.5.1 
Image ID:      quay.io/oauth2-proxy/oauth2-proxy@sha256:3c6012940d3cd71ecc67fd7783916df6d27972b520209af36e9567b9bdbc0c79
```

GitLab Enterprise 16+

**Comment by tuunit** [comment]
Working on a fix: https://github.com/oauth2-proxy/oauth2-proxy/pull/2299
Already tested locally and it resolves this issue.

**Comment by tuunit** [comment]
Resolved by #2299.
